redash/handlers/query_results.py

```diff
@@ -182,7 +182,12 @@ def post(self, query_id):
         if has_access(query, self.current_user, allow_executing_with_view_only_permissions):
             return run_query(query.parameterized, parameter_values, query.data_source, query_id, max_age)
         else:
-            return {'job': {'status': 4, 'error': 'You do not have permission to run queries with this data source.'}}, 403
+            if current_user.is_api_user() and not query.parameterized.is_safe:
+                message = 'This query contains potentially unsafe parameters and cannot be executed on a publicly shared resource.'
```

Comment:
```suggestion
                message = 'This query contains potentially unsafe parameters and cannot be executed on a shared dashboard or visualization embed.'
```

Reply:
I deliberately used "resource" to avoid the word "dashboard" appearing on embeds and vice versa, but 🤷🏻‍♂️

Reply:
I know, but "publicly shared resource" sounds like something taken from a programming manual and not very friendly. :)

Reply:
![image](https://user-images.githubusercontent.com/289488/61353763-3d08b200-a879-11e9-9ffd-cb051046fe17.png)